tracing/providers/opentracing/src/main/java/io/helidon/tracing/providers/opentracing/OpenTracingSpan.java

```diff
@@ -90,13 +97,27 @@ public void end(Throwable throwable) {
         delegate.log(Map.of("event", "error",
                             "error.kind", "Exception",
                             "error.object", throwable,
-                            "message", throwable.getMessage()));
+                            "message", throwable.getMessage() != null ? throwable.getMessage() : "none"));
         delegate.finish();
+        spanLifeCycleListeners.forEach(listener -> listener.ended(limited(), throwable));
     }
 
     @Override
     public Scope activate() {
-        return new OpenTracingScope(tracer.activateSpan(delegate));
+        var result = new OpenTracingScope(this, tracer.activateSpan(delegate), spanLifeCycleListeners);
+        UnsupportedActivationException ex = new UnsupportedActivationException("Error activating span", result);
```

Comment:
See earlier comment about heavyweight exception creation; would be nice to avoid this if at all possible

Reply:
done